pkg/operator/ceph/cluster/osd/update.go

```diff
@@ -178,7 +178,7 @@ func (c *updateConfig) updateExistingOSDs(errs *provisionErrors) {
 			updateConditionFunc(c.cluster.clusterInfo.Context, c.cluster.context, c.cluster.clusterInfo.NamespacedName(), k8sutil.ObservedGenerationNotAvailable, cephv1.ConditionProgressing, v1.ConditionTrue, cephv1.ClusterProgressingReason, message)
 		}
 		if err != nil {
-			errs.addError("%v", errors.Wrapf(err, "failed to update OSD %d", osdID))
+			errs.addError("%v", errors.Wrapf(err, "failed to update OSD %d - DeviceSets may have changed", osdID))
```

Comment:
Rather than updating this overall error, how about the more specific inner message [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/osd/osd.go#L422)? That location should be much more specific to the device sets changing.

Reply:
Is this really possible that a device set can be changed during a update osd?

Reply:
The device set can be updated in the cluster CR between different reconciles

Reply:
Got it thanks,
I thought it was referring to device-class